bash-config: /etc/profile.d cleanup survey

leedagee opened the pull request:
<!-- For description on topic creation and maintenance, please refer to [this Wiki article](https://wiki.aosc.io/developer/packaging/topic-based-maintenance-guideline/). -->

Topic Description
-----------------

[bash-config#23](https://github.com/AOSC-Dev/bash-config/pull/23) fixed a bash initialization related bug, while it reveals a wide range of flaws in our packaging work. This topic moves the misused profile.d scripts to correct directories for a consistent behavior.

Package(s) Affected
-------------------

Currently, these package provides /etc/profile.d:

- [x] ~32subsystem~
- [x] ~apache-ant~
- [x] ~apache-maven~
- [x] ~arduino-mk~
- [x] autojump-rs
- [x] ~bash-completion~
- [x] ~bash-startup~
- [x] ~byobu~
- [x] ~cuda~
- [x] dotnet-host
- [x] ~elfutils~
- [x] ~firefox~
- [x] ~flatpak~
- [x] fzf
- [x] ~gawk~
- [x] gpm (help wanted, seems ok)
- [x] ~gradle~
- [x] grc (placed in profile.d by upstream, the broken feature is not widely used, moved to bashrc.d)
- [x] ~hook-uname~ (~help wanted, why do oma provides output this~, /etc/profile.d is dropped by 68ec4845b)
- [x] ~ksshaskpass~
- [x] ~libreoffice~
- [x] ~mercurial~
- [x] ~midori~
- [x] ~mingw+w64~
- [x] ~mozilla-common~
- [x] nvm
- [x] oma (invalid, discuss in oma topic)
- [x] ~openjdk~
- [x] ~openjdk-11~
- [x] ~openjdk-8~
- [x] ~perl~
- [x] powershell
- [x] ~qt-5~
- [x] ruby
- [x] ~seahorse~
- [x] ~snapd~
- [x] ~stackless-env~ (~help wanted~, drop orphaned)
- [x] ~thunderbird~
- [x] udisks 
- [x] vte
- [x] wezterm
- [x] ~x11-ssh-askpass~

Overall, changes actually applied for:
- udisks
- ~powershell~
- autojump-rs
- fzf
- grc
- wezterm
- dotnet-host
- nvm
- ruby
- vte

(Strike-through means the package uses /etc/profile.d correctly, this survey topic is not applicable.)

<!-- Please list all package(s) affected by this topic here. -->

Security Update?
----------------

<!-- If this topic contains security update(s), please uncomment "Yes,"
     mark with the `security` and `priority` labels, and make sure to mark your commits to relevant issue numbers.

     Please see GitHub's documentation on "Linking a pull request to an issue":

     https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue -->

<!-- Yes - Issue Number: ISSUENUMBER -->
No 

<!-- Please uncomment the "Build Order" section if your topic affects more than one package. -->

Test Build(s) Done
------------------

**Primary Architectures**

<!-- Please remove any architecture to which this topic does not apply. -->

- [ ] AMD64 `amd64`   
- [ ] AArch64 `arm64`
<!-- If this package involves a `+32` counterpart, please uncomment the line below. -->    
<!-- - [ ] 32-bit Optional Environment `optenv32` -->

<!-- If all package(s) affected by this topic is `noarch`, please use the stub below. -->
<!-- - [ ] Architecture-independent `noarch` -->

**Secondary Architectures**

<!-- Please remove any architecture to which this topic does not apply. -->

Architectural progress for "secondary," or experimental ports does not impede on merging of this topic.

- [ ] Loongson 3 `loongson3`
- [ ] MIPS R6 64-bit (Little Endian) `mips64r6el`
- [ ] PowerPC 64-bit (Little Endian) `ppc64el`
- [ ] RISC-V 64-bit `riscv64`

<!-- Maintainers should review file changes and, if the build script(s) affected complies with the
     [Package Styling Manual](https://wiki.aosc.io/developer/packaging/package-styling-manual/), please
     add `lgtm` label to this issue. -->

<!-- Maintainers and users may now test the packages in this topic and, once user/maintainer feedback indicates that the update(s) work as expected and find its quality satisfactory,
     another maintainer may now review this pull request and mark it as Approved. After which, the maintainer will build affected package(s) and upload them to the `stable` repository. -->

Update(s) Uploaded to Stable
----------------------------

**Primary Architectures**

<!-- Please remove any architecture to which this topic does not apply. -->

- [ ] AMD64 `amd64`   
- [ ] AArch64 `arm64`
<!-- If this package involves a `+32` counterpart, please uncomment the line below. -->
<!-- - [ ] 32-bit Optional Environment `optenv32` -->

<!-- If all package(s) affected by this topic is `noarch`, please use the stub below. -->
<!-- - [ ] Architecture-independent `noarch` -->

**Secondary Architectures**

<!-- Please remove any architecture to which this topic does not apply. -->

Architectural progress for "secondary," or experimental ports does not impede on merging of this topic.

- [ ] Loongson 3 `loongson3`
- [ ] MIPS R6 64-bit (Little Endian) `mips64r6el`
- [ ] PowerPC 64-bit (Little Endian) `ppc64el`
- [ ] RISC-V 64-bit `riscv64`

<!-- TODO: CI to auto-fill architectural progress. -->

leedagee committed:
nvm: move 30-nvm.sh to /etc/bashrc.d/

leedagee committed:
dotnet-host: set environment variable instead of shell variable

leedagee committed:
powershell: export environment variable

leedagee committed:
ruby: export PATH in /etc/profile.d/ruby-gem.sh

leedagee committed:
autojump-rs: move autojump.bash symlink to bashrc.d

leedagee committed:
fzf: move fzf-key-bindings.* to corresponding dirs

cthbleachbit commented on lang-js/nvm/autobuild/build:
```diff
@@ -1,13 +1,15 @@
-install -d "${PKGDIR}/usr/share/doc/nvm/"
-cp *.md "${PKGDIR}/usr/share/doc/nvm/"
+abinfo "Installing nvm shell startup files, bash completion and doc..."
+install -Dvm644 nvm_aosc.sh "${PKGDIR}/etc/bashrc.d/30-nvm.sh"
```
Any rationale on why this export should be a bash specifc thing? Looks like NVM_DIR should be set regardless of default shell choice.

Also note that desktop environments generally don't source `/etc/bashrc.d` but would include `/etc/profile.d`. If this environment variable were to be used by any graphical applications (say, an IDE or a graphical text editor) they will no longer see `NVM_DIR` in their environment.

cthbleachbit commented on lang-js/nvm/autobuild/build:
```diff
@@ -1,13 +1,15 @@
+install -Dvm644 nvm_aosc.sh "${PKGDIR}/etc/bashrc.d/30-nvm.sh"
+install -Dvm644 bash_completion "${PKGDIR}/usr/share/bash-completion/completions/nvm"
+install -dv "${PKGDIR}/etc/zsh/zshrc.d"
+ln -vs ../../bashrc.d/30-nvm.sh "${PKGDIR}/etc/zsh/zshrc.d/30-nvm"
```
Just leave it in `/etc/profile.d`...

leedagee committed:
nvm: rewrite build script

Reason:
- nvm.sh itself is not a configuration file. Its content should not be
  part of the conffiles.
- NVM_DIR should be set only once, not every time the user opens a shell.
  This also give user the ability to customize NVM_DIR (with a `nvm
  unload' command before reset NVM_DIR)
- Sourcing nvm.sh causes two effects: the PATH and MANPATH environment
  variables are updated, the CURRENT shell gets the nvm cli tool. It is
  required to run nvm.sh both in profile and shellrc.

Rewritten nvm package build script puts original nvm.sh directly to
/usr/share/nvm without the NVM_DIR stub. Instead, define NVM_DIR in
/etc/profile.d/30-nvm and immediately source nvm.sh to have PATH and
MANPATH correctly set. For interactive shells, we can just source the
original nvm.sh for the cli tool.

MingcongBai committed:
powershell: rebuild for OpenSSL 3.1.1

MingcongBai committed:
powershell: update to 7.3.6; clean up

- Improve PKGDES.
- Use bundled .NET SDK during build time (Kexy!).
- Lint scripts in accordance with the Styling Manual.

leedagee committed:
udisks: bash completion for non-login bash

leedagee committed:
grc: automatic aliases for non-login shells

leedagee committed:
vte: shell integrations for non-login ones

MingcongBai commented on lang-js/nvm/autobuild/build:
```diff
@@ -1,13 +1,10 @@
-cp *.md "${PKGDIR}/usr/share/doc/nvm/"
+abinfo "Creating symlink to nvm.sh in bashrc.d and zshrc.d"
+mkdir -pv "${PKGDIR}"/etc/{bashrc.d,zsh/zshrc.d}
+ln -vs /usr/share/nvm/nvm.sh "${PKGDIR}/etc/bashrc.d/30-nvm"
```
Use relative paths here.

MingcongBai commented on lang-js/nvm/autobuild/build:
```diff
@@ -1,13 +1,10 @@
-abinfo "Creating nvm profile..."
+abinfo "Installing nvm.sh, bash completion and doc..."
+install -Dvm644 nvm.sh "${PKGDIR}/usr/share/nvm/nvm.sh"
```
Missing absolute path "$SRCDIR"/...

MingcongBai commented on lang-js/nvm/autobuild/build:
```diff
@@ -1,13 +1,10 @@
+install -Dvm644 nvm.sh "${PKGDIR}/usr/share/nvm/nvm.sh"
+install -Dvm644 bash_completion "${PKGDIR}/usr/share/bash-completion/completions/nvm"
+mkdir -pv "${PKGDIR}/usr/share/doc/nvm/"
+cp -v *.md "${PKGDIR}/usr/share/doc/nvm/"
```
... Until here.

MingcongBai replied:
Also, you could have just used `install -Dvm644 "$SRCDIR"/*.md -t "$PKGDIR"/usr/share/doc/nvm/`.

leedagee committed:
nvm: several fixes in the build script

leedagee committed:
vte: fix symbolic link for zshrc.d

leedagee committed:
udisks: fix build error

leedagee commented:
powershell is cherry picked from openssl-3.3
wezterm is left unchanged
